### PR TITLE
[AVCe, HEVCe] Fix for default MaxQPI, MaxQPP, MaxQPB

### PR DESCRIPTION
**Pull request by ogolikov**
Issue: MDP-50776
Test:  manually (behavior is needed)

### CODE REVIEW

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp [hunk 1]
```diff
@@ -5703,6 +5703,16 @@ void MfxHwH264Encode::SetDefaults(
     if (extOpt2->RepeatPPS == MFX_CODINGOPTION_UNKNOWN)
         extOpt2->RepeatPPS = MFX_CODINGOPTION_ON;
 
+    if (extOpt2->MinQPI || extOpt2->MinQPP || extOpt2->MinQPB ||
+        extOpt2->MaxQPI || extOpt2->MaxQPP || extOpt2->MaxQPB)
+    {
+        if (!extOpt2->MaxQPI)
```
Comment (mgonchar) [inline _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: why don't you change mins as well?

File: _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp [hunk 2]
```diff
@@ -2908,6 +2908,23 @@ void SetDefaults(
     if (IsOff(CO3.EnableQPOffset))
         Zero(CO3.QPOffset);
 
+    if (CO2.MinQPI || CO2.MinQPP || CO2.MinQPB ||
```
Comment (mgonchar) [inline _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp]: why do we need this condition?

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp (hunk 1 above)
Comment (ogolikov) [reply _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: question isn't clear for me

Comment (mgonchar) [reply _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: I mean don't we need following code
```
if (!extOpt2->MinQPI)
    extOpt2->MinQPI = 1;
...
```

File: _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp (hunk 2 above)
Comment (lakulako) [reply _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp]: Don't refer internal tools.
Most likely "if any is initialized, other should be initialized too".

File: _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp [hunk 3]
```diff
@@ -5703,6 +5703,16 @@ void MfxHwH264Encode::SetDefaults(
...
+           extOpt2->MaxQPI = 51;
```
Comment (lakulako) [inline _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: Will it work for >8bit samples?

Comment (ogolikov) [reply _studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp]: there is no >8bits support in h264 encoder

**Comment by mgonchar** [comment]
@ogolikov the question is still there, why don't you set minimum QP to 1 inside ```void MfxHwH264Encode::SetDefaults``` ? You did it for hevc

**Comment by mgonchar** [comment]
ok, I checked driver code, so they are clamping QPs to range [1, 51], see https://github.com/intel/media-driver/blob/bc5190beb1f538f239b00ba3b4aa222fe7f4ee7d/media_driver/agnostic/common/codec/hal/codechal_vdenc_avc.cpp#L3212

**Pull request merged**